src/RandomNumberSeedService.cxx

```diff
@@ -0,0 +1,89 @@
+#include "Event/EventHeader.h"
+#include "Event/RunHeader.h"
+#include "Framework/Process.h"
+#include "Framework/RandomNumberSeedService.h"
+#include <time.h>
+
+std::ostream& operator<<(std::ostream& s, const ldmx::RandomNumberSeedService& o) {
+  o.stream(s);
+  return s;
+}
+
+
+namespace ldmx {
+
+static const int SEED_EXTERNAL = 2;
+static const int SEED_RUN =  3;
+static const int SEED_TIME = 4;
+
+void RandomNumberSeedService::stream(std::ostream& s) const {
+  s << "RandomNumberSeedService(";
+  if (seedMode_==SEED_RUN) s << "Seed on RUN";
+  if (seedMode_==SEED_TIME) s << "Seed on TIME";
+  if (seedMode_==SEED_EXTERNAL) s << "Seeded EXTERNALLY";
+  s << ") Master seed = " <<masterSeed_ << std::endl;
+  for (auto i: seeds_) {
+    s << " " << i.first << "=>" << i.second << std::endl;
+  }
+}
+
+
+RandomNumberSeedService::RandomNumberSeedService(const std::string& name, const std::string& tagname, const Parameters& parameters, Process& process) : ConditionsObject(CONDITIONS_OBJECT_NAME), ConditionsObjectProvider(CONDITIONS_OBJECT_NAME,tagname,parameters,process) {
+  std::string seeding=parameters.getParameter<std::string>("seedMode","run");
+  if (!strcasecmp(seeding.c_str(),"run")) {
+    seedMode_=SEED_RUN;
+  }
+  if (!strcasecmp(seeding.c_str(),"external")) {
```

Comment:
Why is this a block of 3 if statements instead of just an if - if else else block.  Only one mode is possible at a time correct?